src/WrappedMToken.sol

```diff
@@ -454,38 +503,28 @@ contract WrappedMToken is IWrappedMToken, Migratable, ERC20Extended {
     function _transfer(address sender_, address recipient_, uint240 amount_, uint128 currentIndex_) internal {
         _revertIfInvalidRecipient(recipient_);
 
-        // Claims for both the sender and recipient are required before transferring since add an subtract functions
-        // assume accounts' balances are up-to-date with the current index.
         _claim(sender_, currentIndex_);
         _claim(recipient_, currentIndex_);
 
         emit Transfer(sender_, recipient_, amount_);
 
         if (amount_ == 0) return;
 
-        Account storage senderAccountInfo_ = _accounts[sender_];
-        Account storage recipientAccountInfo_ = _accounts[recipient_];
+        if (sender_ == recipient_) {
+            uint240 balance_ = _accounts[sender_].balance;
 
-        // If the sender and recipient are both earning or both non-earning, update their balances without affecting
-        // the total earning and non-earning supply storage variables.
-        if (senderAccountInfo_.isEarning == recipientAccountInfo_.isEarning) {
-            uint240 senderBalance_ = senderAccountInfo_.balance;
-
-            if (senderBalance_ < amount_) revert InsufficientBalance(sender_, senderBalance_, amount_);
-
-            unchecked {
-                senderAccountInfo_.balance = senderBalance_ - amount_;
-                recipientAccountInfo_.balance += amount_;
-            }
+            if (balance_ < amount_) revert InsufficientBalance(sender_, balance_, amount_);
 
             return;
         }
 
-        senderAccountInfo_.isEarning
+        // TODO: Don't touch globals if both are earning or not earning.
```

Comment:
optimization mentioned in TODO?

Reply:
In a future PR, if needed at all. May depend on other PRs as well.